[v12] Enforce body size limits for http responses

jentfoo opened the pull request:
v12 backport of PR https://github.com/gravitational/teleport/pull/33768

Similar to #33768, the `e` reference is being updated here.  However in this we are pointing to the merged `e` PR: https://github.com/gravitational/teleport.e/pull/2495

zmb3 commented on e:
Similarly, might as well update to latest while you're here.

jentfoo committed:
Enforce body size limits for http responses (#33768)

This change enforces limits on the response size for http requests across Teleport.
The `e` reference is also being updated as the referenced commit includes similar fixes: https://github.com/gravitational/teleport.e/pull/2495
